test/document_stores/test_faiss.py

```diff
@@ -8,294 +9,266 @@
-
-    # test loading the index via init
-    new_document_store = FAISSDocumentStore(
-        faiss_index_path=tmp_path / "haystack_test_faiss", faiss_config_path=tmp_path / "custom_path.json"
-    )
-
-    # check faiss index is restored
-    assert new_document_store.faiss_indexes[document_store.index].ntotal == len(DOCUMENTS)
-    # check if documents are restored
-    assert len(new_document_store.get_all_documents()) == len(DOCUMENTS)
-    # Check if the init parameters are kept
-    assert not new_document_store.progress_bar
-
-
-@pytest.mark.skipif(sys.platform in ["win32", "cygwin"], reason="Test with tmp_path not working on windows runner")
-def test_faiss_index_mutual_exclusive_args(tmp_path):
-    with pytest.raises(ValueError):
-        FAISSDocumentStore(
-            sql_url=f"sqlite:////{tmp_path/'haystack_test.db'}",
-            faiss_index_path=f"{tmp_path/'haystack_test'}",
-            isolation_level="AUTOCOMMIT",
-        )
+class TestFAISSDocumentStore(DocumentStoreBaseTestAbstract):
+    index_name = __name__
 
-    with pytest.raises(ValueError):
-        FAISSDocumentStore(
-            f"sqlite:////{tmp_path/'haystack_test.db'}",
-            faiss_index_path=f"{tmp_path/'haystack_test'}",
+    @pytest.fixture
+    def ds(self, tmp_path):
+        return FAISSDocumentStore(
+            sql_url=f"sqlite:///{tmp_path}/haystack_test.db",
+            return_embedding=True,
+            index=self.index_name,
             isolation_level="AUTOCOMMIT",
+            progress_bar=False,
+            similarity="cosine",
         )
 
+    @pytest.fixture(scope="class")
+    def documents_embedding_only(self, documents):
+        # drop documents without embeddings from the original fixture
+        return [d for d in documents if d.embedding is not None]
+
+    @pytest.mark.unit
+    def test_index_mutual_exclusive_args(self, tmp_path):
+        with pytest.raises(ValueError):
+            FAISSDocumentStore(
+                sql_url=f"sqlite:////{tmp_path/'haystack_test.db'}",
+                faiss_index_path=f"{tmp_path/'haystack_test'}",
+                isolation_level="AUTOCOMMIT",
+            )
+
+        with pytest.raises(ValueError):
+            FAISSDocumentStore(
+                f"sqlite:////{tmp_path/'haystack_test.db'}",
+                faiss_index_path=f"{tmp_path/'haystack_test'}",
+                isolation_level="AUTOCOMMIT",
+            )
+
+    @pytest.mark.integration
+    def test_delete_index(self, ds, documents):
+        """Contrary to other Document Stores, FAISSDocumentStore doesn't raise if the index is empty"""
+        ds.write_documents(documents, index="custom_index")
+        assert ds.get_document_count(index="custom_index") == len(documents)
+        ds.delete_index(index="custom_index")
+        assert ds.get_document_count(index="custom_index") == 0
+
+    @pytest.mark.integration
+    def test_index_save_and_load(self, ds, documents_embedding_only, tmp_path):
```

Comment:
This and the following one can be compressed into a single parametrized one, given that only the path where the config is saved changes